frontend/wallet/src/main/DaemonProcess.re

```diff
@@ -1,40 +1,101 @@
 open Bindings;
 
-let start = port => {
-  print_endline("Starting graphql-faker");
-  let p =
-    ChildProcess.spawn(
-      "node",
-      [|
-        Filename.concat(
-          ProjectRoot.resource,
-          "node_modules/graphql-faker/dist/index.js",
-        ),
-        "--port",
-        string_of_int(port),
-        "--",
-        "schema.graphql",
-      |],
-    );
+module Command = {
+  type t = {
+    executable: string,
+    args: array(string),
+    env: Js.Dict.t(string),
+  };
+};
+
+module Process = {
+  let start = (command: Command.t, port) => {
+    print_endline("Starting graphql-faker");
 
-  ChildProcess.Process.onError(p, e =>
-    prerr_endline(
-      "Daemon process crashed: " ++ ChildProcess.Error.messageGet(e),
-    )
-  );
-
-  ChildProcess.Process.onExit(p, (n, s) =>
-    if (n == 1) {
-      Printf.fprintf(
-        stderr,
-        "Daemon process exited with non-zero exit code: Exit:%d, msg:%s%!\n",
-        n,
-        Tc.Option.withDefault(s, ~default="Port 8080 already in use?"),
+    let p =
+      ChildProcess.spawn(
+        command.executable,
+        command.args,
+        {"env": command.env},
       );
-    } else {
-      print_endline("Shutting down daemon process.");
-    }
-  );
 
-  () => ChildProcess.Process.kill(p);
+    ChildProcess.Process.onError(p, e =>
+      prerr_endline(
+        "Daemon process "
+        ++ command.executable
+        ++ " crashed: "
+        ++ ChildProcess.Error.messageGet(e),
+      )
+    );
+
+    ChildProcess.ReadablePipe.on(ChildProcess.Process.stderrGet(p), "data", s =>
+      prerr_endline(
+        "Daemon "
+        ++ command.executable
+        ++ " error: "
+        ++ Node.Buffer.toString(s),
+      )
+    );
+
+    ChildProcess.Process.onExit(p, (n, s) =>
+      if (n == 1) {
+        Printf.fprintf(
+          stderr,
+          "Daemon process exited with non-zero exit code: Exit:%d, msg:%s%!\n",
+          n,
+          Tc.Option.withDefault(
+            s,
+            ~default="Port " ++ Js.Int.toString(port) ++ " already in use?",
+          ),
+        );
+      } else {
+        print_endline("Shutting down daemon process.");
+      }
+    );
+
+    () => ChildProcess.Process.kill(p);
+  };
+};
+
+let (^/) = Filename.concat;
+
+let startAll = (~fakerPort, ~codaPort) => {
+  let graphqlFaker = {
+    Command.executable: "/run/current-system/sw/bin/node",
```

Comment:
You can probably avoid this if you don't wipe the path? Could make the env optional or recreate the PATH var